tests/e2e/sanitizer_test.go

```diff
@@ -0,0 +1,398 @@
+
+		if err != nil {
+			return err
+		}
+
+		lineNum, err := strconv.Atoi(lineNum1Value)
+
+		if err != nil {
+			return fmt.Errorf("could not convert number %s: %s", lineNum1Value, err)
+		}
+
+		sanitizedFileChangeLines[lineNum] = expectedInsertedValue
+	}
+	return nil
+}
+
+func validE2EProcess(t *testing.T, webDriver selenium.WebDriver) error {
+	var err error
+	t.Log("Starting validating e2e process...\n")
+	filesToSanitize := make([]string, len(sanitizeFileToExpectedSanitation)+len(untouchedFileNames))
+
+	i := 0
+	for fileToSanitize := range sanitizeFileToExpectedSanitation {
+		filesToSanitize[i] = fileToSanitize
+		i++
+	}
+	for index := range untouchedFileNames {
+		filesToSanitize[i] = untouchedFileNames[index]
+		i++
+	}
+
+	//Wait for page to be ready
+	err = WaitForUploadButtonIsReady(webDriver, MaxWaitTimeout)
+	if err != nil {
+		return err
+	}
+	t.Log("Uploading File")
+	err = UploadFiles(webDriver, filesToSanitize)
+	if err != nil {
+		return err
+	}
+
+	err = WaitForUntouchedFilesReady(webDriver, MaxWaitTimeout)
+
+	if err != nil {
+		return err
+	}
+
+	// Verify Untouched Files
+	t.Logf("Verifying untouched files: %s", untouchedFileNames)
+	unsanitizedElements, err := webDriver.FindElements(selenium.ByID, "unsanitized_file_p")
+
+	if err != nil {
+		return err
+	}
+	assert.Equal(t, len(untouchedFileNames), len(unsanitizedElements), "Expected untouched file names to be %d, but got %d", len(untouchedFileNames), unsanitizedElements)
+	for _, unsanitizedElement := range unsanitizedElements {
+		actualUnsanitizedFileName, err := unsanitizedElement.Text()
+		if err != nil {
+			return err
+		}
+
+		assert.True(t, slices.Contains(untouchedFileNames, actualUnsanitizedFileName), "Unexpected untouched file '%s' found. Expected untouched files: %s", actualUnsanitizedFileName, untouchedFileNames)
+	}
+
+	values, err := webDriver.FindElements(selenium.ByCSSSelector, ".d2h-file-wrapper")
+	if err != nil {
+		return err
+	}
+	fmt.Println("len of sanitized_diff_div", len(values))
+	for _, fileDivElement := range values {
+		err = verifyFileSanitation(t, fileDivElement)
+		if err != nil {
+			return err
+		}
+	}
+
+	// Check download
+	downloadButton, _ := webDriver.FindElement(selenium.ByID, "download_button_label")
+	err = downloadButton.Click()
+	if err != nil {
+		return err
+	}
+	time.Sleep(25 * time.Second)
+
+	for sanitizedFileName := range changedLines {
+		expectedFileToDownloadPath := filepath.Join(DownloadPath, sanitizedFileName)
+		_, downloadedFileError := os.Stat(expectedFileToDownloadPath)
+		assert.Nil(t, downloadedFileError, downloadedFileError)
+		err = verifyDownloadedFile(t, expectedFileToDownloadPath, changedLines[sanitizedFileName])
+		if err != nil {
+			return err
+		}
+	}
+
+	// View Rules
+	err = verifyViewRules(t, webDriver)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func verifyDownloadedFile(t *testing.T, filename string, changedLines map[int]string) error {
```

Comment:
Ensure that the output file size is an expected value. This will catch any metadata/header changes.
You can use os.Stat(filePath) to get a FileInfo struct that has the size value.